frontend/app/admin/reports/report.tsx

```diff
@@ -0,0 +1,227 @@
+
+      const data = await response.json();
+
+      if (data.error) {
+        setError(data.error);
+        throw data.error;
+      }
+
+      setAllReports(reports =>
+        reports.map(report => {
+          if (report.targetUserId === data.id && report.targetUser) {
+            return { ...report, targetUser: { ...report.targetUser, isActive: data.isActive } };
+          }
+          return report;
+        }),
+      );
+      setError("");
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  const handleCheckBox = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    try {
+      await markReportDone(report.id, e.target.checked);
+      setError("");
+    } catch (error) {
+      setError("Internal server error");
+    }
+  };
+  const showDate = (date: Date) => {
+    const dateFormatted = date.toISOString().slice(0, 10).split("-").reverse().join(".");
+    return dateFormatted;
+  };
+  const handleDeleteReportSubmit = async () => {
+    try {
+      const response = await deleteReportById(report.id);
+      setAllReports(reports => reports.filter(report => report.id !== response.id));
+      setError("");
+    } catch (error) {
+      setError("Internal server error");
+    }
+  };
+  return (
+    <div className="admin-panel-reports-grid">
+      <div>
+        <label className="admin-panel-report-label">Done</label>
+        <input
+          type="checkbox"
+          id="checkbox"
+          defaultChecked={report.done}
+          onChange={e => handleCheckBox(e)}
+        />
+      </div>
+      <div>
+        <label className="admin-panel-report-label">Reporter</label>
+        <Link href={`/users/${report.creatorId}`}>{report.creator?.username}</Link>
+
+        <p className={report.creator?.isActive ? "admin-panel-status-active" : "admin-panel-status-suspended"}>
+          {report.creator?.isActive
+            ? "Active"
+            : "On suspension " +
+              (report.creator?.banDuration ? "until " + report.creator.banDuration.toDateString() : "forever")}
+        </p>
+      </div>
+      <div>
+        <label className="admin-panel-report-label">Date</label>
+        <p>{showDate(report.created_at)}</p>
+      </div>
+      <div>
+        <label className="admin-panel-report-label">Target</label>
+        <Link href={`/users/${report.targetUserId}`}>{report.targetUser?.username}</Link>
+
+        <p className={report.targetUser?.isActive ? "admin-panel-status-active" : "admin-panel-status-suspended"}>
+          {report.targetUser?.isActive
+            ? "Active"
+            : "On suspension " +
+              (report.targetUser?.banDuration ? "until " + report.targetUser.banDuration.toDateString() : "forever")}
+        </p>
+      </div>
+      <div className="admin-panel-report-content description">
+        <label className="admin-panel-report-label">Description</label>
+        <p>{report.content}</p>
+      </div>
+
+      {report.targetUser?.role !== Role.admin ? (
+        <div>
```

Comment:
Jos buttonit käärii diviin, jolla on tämä class, ne pysyy pienemmillä näkymillä aina rinnakkain eikä hypi välillä päällekkäin.

```
 @media only screen and (max-width: abstracts.$large-width) {
    .report-buttons {
      display: inline-flex;
      gap: 4px;

      * {
        margin: 0;
        min-width: fit-content;
      }
    }
}
```